### PR TITLE
Added password recovery feature

### PR DESCRIPTION
**Pull request by nishithshowri006**

### CODE REVIEW

**Comment by nishithshowri006** [comment]
@akshgpt7 I didn't add any mail features. When the password reset link will be appeared with metadata in a folder called sent_emails

**Comment by akshgpt7** [comment]
Great work, just add the change password functionality as well and then I'll merge it!
About the email, I asked the mods and they suggest to use a spare email. I've asked in our team channel.

**Comment by nishithshowri006** [comment]
@akshgpt7 added the functionality and it is working but I didn't add any html

File: annoyed-alligators/socl_media/urls.py [hunk 1]
```diff
@@ -21,9 +21,29 @@
...
+    path('password-reset/', auth_views.PasswordResetView.as_view(
+        template_name='users/password_reset.html'), name="password_reset"),
+    path('password-reset/done/', auth_views.PasswordResetDoneView.as_view(
+        template_name='users/password_reset_done.html'),
+        name="password_reset_done"),
+    path('password-reset-confirm/<uidb64>/<token>/',
+         auth_views.PasswordResetConfirmView.as_view(
+          template_name='users/password_reset_confirm.html'),
+         name="password_reset_confirm"),
+    path('password_reset_complete',
```
Comment (akshgpt7) [inline annoyed-alligators/socl_media/urls.py]: Don't add underscores in url path. Change these to hyphens as in other paths.

**Review by akshgpt7 — CHANGES_REQUESTED** [review]

**Comment by akshgpt7** [comment]
Awesome! Just one thing: **Please add type hints wherever possible and docstrings for every view. Also, add comments in your code to explain the functioning (There are points for these things).**

Also, can you add the spare email address for password reset?

File: annoyed-alligators/socl_media/apps/feed/models.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 
 class Post(models.Model):
+
```
Comment (akshgpt7) [inline annoyed-alligators/socl_media/apps/feed/models.py:8]: Add docstrings for models as well, telling what they store.

File: annoyed-alligators/socl_media/apps/feed/views.py [hunk 3]
```diff
@@ -7,6 +7,12 @@
 
 @login_required
 def feed(request):
```
Comment (akshgpt7) [inline annoyed-alligators/socl_media/apps/feed/views.py]: For this, there's a built in Django ListView. It's a class based view and does most of the work internally. Please use the built-in view. 
Here's how to use them: https://youtu.be/Xeh9r0CXBmU

**Review by akshgpt7 — CHANGES_REQUESTED** [review]

**Comment by akshgpt7** [comment]
Also, can you add the email part?

**Comment by nishithshowri006** [comment]
I actually added email part but the link at present is redirected into txt file in our project folder. We need to use some third party tools for sending email to user

**Pull request merged**